src/CoapClient.ts

```diff
@@ -516,11 +574,52 @@ export class CoapClient {
 	private static send(
 		connection: ConnectionInfo,
 		message: Message,
+		highPriority: boolean = false,
 	): void {
 
-		// send the message
-		connection.socket.send(message.serialize(), connection.origin);
+		// Put the message in the queue
+		if (highPriority) {
+			// in front
```

Comment:
We should use 2 queues, 1 high-priority and 1 low-priority. Or a counter to know how many high-prio messages there are.